ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto

```diff
@@ -223,32 +223,17 @@ message DisclosedContract {
   // Required
   oneof arguments {
     // The contract arguments as typed Record
-    Record create_arguments = 3;
+    // Deprecated in favor of `contract_create_data`
+    Record create_arguments = 3 [deprecated = true];
 
-    // The contract arguments specified using an opaque blob extracted from the ``create_arguments_blob`` field
-    // of a ``com.daml.ledger.api.v1.CreatedEvent``.
-    google.protobuf.Any create_arguments_blob = 5;
+    // Opaque byte string containing the complete payload required by the Daml engine
+    // to reconstruct a contract not known to the receiving participant.
+    // Required
+    google.protobuf.Any contract_create_payload = 5;
   }
 
   // The contract metadata from the create event.
-  // Required
-  ContractMetadata metadata = 4;
-
-  // The signatories for this contract as specified by the template.
-  // Required
-  repeated string signatories = 6;
-
-  // The observers for this contract as specified explicitly by the template or implicitly as choice controllers.
-  // This field never contains parties that are signatories.
-  // Optional
-  repeated string observers = 7;
-
-  // The key of the created contract.
-  // This will be set if and only if ``create_arguments`` is set and ``template_id`` defines a contract key.
-  // Optional
-  Value contract_key = 8;
-
-  // Maintainers of the contract key, if defined.
+  // Deprecated in favor of `contract_create_data`
   // Optional
-  repeated string key_maintainers = 9;
+  ContractMetadata metadata = 4 [deprecated = true];
```

Comment:
I'm not a fan of the added delivery risk of removing `ContractMetadata`. In particular the LET stored in there is very useful for clients.

I'd suggest to move this PR to a state where no encoding changes are required, and do the cleanup (if needed at all in 3.0).

Reply:
Deprecating this field is not risky as long as the code underneath is untouched. It merely sends a signal to the clients to beware

Reply:
ack

Reply:
@meiersi-da - wasn't the outcome of Monday's meeting that going forward clients will need to store any useful data in a wrapper message ?

Reply:
I intend to keep the metadata deprecation in 2.8